src/utils/trust.details.ts

```diff
@@ -169,67 +195,68 @@ export const postTrustDetails = async (req: Request, res: Response, next: NextFu
     if (!errorList.isEmpty() || errors.length) {
       const errorListArray = !errorList.isEmpty() ? errorList.array() : [];
       const formattedErrors = formatValidationError([...errorListArray, ...errors]);
-
       let pageProps;
       if (req.query["relevant-period"] === "true") {
         pageProps = await getPagePropertiesRelevantPeriod(req, formData, isUpdate, isReview, formattedErrors);
       } else {
         pageProps = await getPageProperties(req, formData, isUpdate, isReview, formattedErrors);
       }
-
       return res.render(pageProps.template, pageProps);
     }
 
-    //  map form data to session trust data
+    // map form data to session trust data
     const details = mapperDetails.mapDetailToSession(req.body, hasNoBoAssignableToTrust(appData));
     if (!details.trust_id) {
       details.trust_id = mapperDetails.generateTrustId(appData);
     }
 
-    //  if present, get existing trust from session (as it might have attached trustees)
-    let trust;
-    if (isReview) {
-      trust = getReviewTrustById(appData, details.trust_id);
-    } else {
-      trust = getTrustByIdFromApp(appData, details.trust_id);
-    }
-    Object.keys(details).forEach(key => trust[key] = details[key]);
+    appData = getAndUpdateTrustData(req, isReview, appData, details);
 
-    //  update trust  in application data at session
-    if (isReview) {
-      updateTrustInReviewList(appData, trust);
-    } else {
-      appData = saveTrustInApp(appData, trust);
-    }
-
-    //  update trusts in beneficial owners
+    // update trusts in beneficial owners
     const selectedBoIds = req.body?.beneficialOwnersIds ?? [];
     appData = updateBeneficialOwnersTrustInApp(appData, details.trust_id, selectedBoIds);
 
-    // // if reviewing a trust, mark trust as in review
+    // if reviewing a trust, mark trust as in review
     if (isReview) {
       setTrustDetailsAsReviewed(appData);
     }
 
-    //  save to session
-    const session = req.session as Session;
+    if (isActiveFeature(config.FEATURE_FLAG_ENABLE_REDIS_REMOVAL) && isRegistration) {
+      await updateOverseasEntity(req, session, appData);
+    } else {
+      await saveAndContinue(req, session);
+    }
     setExtraData(session, appData);
 
-    await saveAndContinue(req, session);
-
     return safeRedirect(res, getNextPage(isUpdate, details.trust_id, req, isReview));
+
   } catch (error) {
     logger.errorRequest(req, error);
-
     return next(error);
   }
 };
 
+const getAndUpdateTrustData = (req: Request, isReview: boolean, appData: ApplicationData, details: Trust) => {
```

Comment:
the name of the parameter details seems odd to me because in reality it is a Trust

Reply:
have now changed it to `trustDetails`..